.github/workflows/publish-packages.yml

```diff
@@ -35,105 +44,37 @@ jobs:
 
       - name: List .tar.gz files
         run: ls -l src/Perlang.ConsoleApp/bin/Release/net6.0/*/*.tar.gz
 
       #
       # Upload files to releases server via rsync
       #
-      - name: Upload linux-arm .tar.gz file to releases server
-        uses: easingthemes/ssh-deploy@v2.1.1
-        env:
-          SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
-          ARGS: "-rltgoDzvO"
-          SOURCE: "./src/Perlang.ConsoleApp/bin/Release/net6.0/linux-arm/*.tar.gz"
-          REMOTE_HOST: ${{ secrets.SSH_REMOTE_HOST }}
-          REMOTE_USER: ${{ secrets.SSH_REMOTE_USER }}
-          TARGET: ${{ secrets.SSH_REMOTE_TARGET }}
-        if: github.ref == 'refs/heads/master'
-
-      - name: Upload linux-arm64 .tar.gz file to releases server
-        uses: easingthemes/ssh-deploy@v2.1.1
-        env:
-          SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
-          ARGS: "-rltgoDzvO"
-          SOURCE: "./src/Perlang.ConsoleApp/bin/Release/net6.0/linux-arm64/*.tar.gz"
-          REMOTE_HOST: ${{ secrets.SSH_REMOTE_HOST }}
-          REMOTE_USER: ${{ secrets.SSH_REMOTE_USER }}
-          TARGET: ${{ secrets.SSH_REMOTE_TARGET }}
-        if: github.ref == 'refs/heads/master'
-
-      - name: Upload linux-x64 .tar.gz file to releases server
-        uses: easingthemes/ssh-deploy@v2.1.1
-        env:
-          SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
-          ARGS: "-rltgoDzvO"
-          SOURCE: "./src/Perlang.ConsoleApp/bin/Release/net6.0/linux-x64/*.tar.gz"
-          REMOTE_HOST: ${{ secrets.SSH_REMOTE_HOST }}
-          REMOTE_USER: ${{ secrets.SSH_REMOTE_USER }}
-          TARGET: ${{ secrets.SSH_REMOTE_TARGET }}
-        if: github.ref == 'refs/heads/master'
-
-      - name: Upload osx-x64 .tar.gz file to build cloud
-        uses: easingthemes/ssh-deploy@v2.1.1
-        env:
-          SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
-          ARGS: "-rltgoDzvO"
-          SOURCE: "./src/Perlang.ConsoleApp/bin/Release/net6.0/osx-x64/*.tar.gz"
-          REMOTE_HOST: ${{ secrets.SSH_REMOTE_HOST }}
-          REMOTE_USER: ${{ secrets.SSH_REMOTE_USER }}
-          TARGET: ${{ secrets.SSH_REMOTE_TARGET }}
-        if: github.ref == 'refs/heads/master'
-
-      - name: Upload win-x64 .tar.gz file to releases server
+      - name: Upload ${{ matrix.arch }} .tar.gz file to releases server
         uses: easingthemes/ssh-deploy@v2.1.1
         env:
           SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
           ARGS: "-rltgoDzvO"
-          SOURCE: "./src/Perlang.ConsoleApp/bin/Release/net6.0/win-x64/*.tar.gz"
+          SOURCE: "./src/Perlang.ConsoleApp/bin/Release/net6.0/${{ matrix.arch }}/*.tar.gz"
           REMOTE_HOST: ${{ secrets.SSH_REMOTE_HOST }}
           REMOTE_USER: ${{ secrets.SSH_REMOTE_USER }}
           TARGET: ${{ secrets.SSH_REMOTE_TARGET }}
         if: github.ref == 'refs/heads/master'
 
+      # TODO: Would ideally run _after_ all jobs of the matrix...
```

Comment:
This is a tricky limitation to get around. There is a `needs` mechanism in GitHub Actions, but it doesn't support propagating all the _files_ (build artifacts) from the jobs to the post-job that way, AFAIK.

I think we'll live with this being executed over and over for now, but the script itself might need to get fixed before we merge this, to avoid creating circular symlinks. (I saw it happening today when running the script manually, with the binaries only for one arch having been uploaded.)

Reply:
The script was adjusted now. (For reference, here's what it looks like - we should perhaps include this in the `git` repo somehow also - the `rm` call is what got added.)

```shell
#!/bin/bash

get_latest_build() {
    PLATFORM=$1
    LATEST_BUILD=$(ls -1 --sort=time /builds/*-${PLATFORM}.tar.gz | head -n1)

    echo $LATEST_BUILD
}

update_symlink() {
    PLATFORM=$1
    LATEST_BUILD=$2

    ln -sf ${LATEST_BUILD} /builds/perlang-latest-${PLATFORM}.tar.gz
}

# Remove old symlinks, to ensure they don't get included in the get_latest_build() calls
rm /builds/perlang-latest*.tar.gz

update_symlink linux-arm $(get_latest_build "linux-arm")
update_symlink linux-arm64 $(get_latest_build "linux-arm64")
update_symlink linux-x64 $(get_latest_build "linux-x64")
update_symlink osx-x64 $(get_latest_build "osx-x64")
update_symlink win-x64 $(get_latest_build "win-x64")
```